internal/primitive/transform/action/strings/split_between_positions.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2023 Linkall Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package strings
+
+import (
+	"fmt"
+
+	"github.com/linkall-labs/vanus/internal/primitive/transform/action"
+	"github.com/linkall-labs/vanus/internal/primitive/transform/arg"
+	"github.com/linkall-labs/vanus/internal/primitive/transform/common"
+	"github.com/linkall-labs/vanus/internal/primitive/transform/context"
+)
+
+type splitBetweenPositionsAction struct {
+	action.CommonAction
+}
+
+// NewSplitBetweenPositionsAction["sourceJSONPath", "startPosition", "endPosition", "targetJsonPath"].
+func NewSplitBetweenPositionsAction() action.Action {
+	return &splitBetweenPositionsAction{
+		CommonAction: action.CommonAction{
+			ActionName: "SPLIT_BETWEEN_POSITIONS",
+			FixedArgs:  []arg.TypeList{arg.EventList, arg.All, arg.All, []arg.Type{arg.EventData}},
+		},
+	}
+}
+
+func (a *splitBetweenPositionsAction) Init(args []arg.Arg) error {
+	a.TargetArg = args[3]
+	a.Args = args[:3]
+	a.ArgTypes = []common.Type{common.String, common.Int, common.Int}
+	return nil
+}
+
+func (a *splitBetweenPositionsAction) Execute(ceCtx *context.EventContext) error {
+	args, err := a.RunArgs(ceCtx)
+	if err != nil {
+		return err
+	}
+
+	v, _ := a.TargetArg.Evaluate(ceCtx)
+	if v != nil {
+		return fmt.Errorf("key %s exists", a.TargetArg.Original())
+	}
+
+	sourceJSONPath, _ := args[0].(string)
+	startPosition, _ := args[1].(int)
+	endPosition, _ := args[2].(int)
+
+	var substrings []string
+
+	if startPosition+endPosition > len(sourceJSONPath) {
```

Comment:
the check is an error, such as the original string length is 10, the start is 5, the end is 8,  5+8 > 10

Reply:
```suggestion
	if startPosition>=endPosition || endPosition > len(sourceJSONPath) {
```

Reply:
As per my understanding, the end position as mentioned in the issue is from the end. 
`"pipeline":[
    {"command":["split_between_positions", "$.data.appinfoA", 2, 2, "$.data. appinfoB"]}
]`
Transformed event data: 
`"appinfoA" : "hello world!",
  "appinfoB" : ["he", "llo worl", "d!"]`
  
  Please correct me If I am wrong.

Reply:
the issue has problem, I have update it

Reply:
The split function targetValue must be an array and length is 3.
* the startPosition  gte endPosition, return an error
* the startPosition gte length of value, the targetValue is []string{value,"",""} 
* the endPosition gte length of value, the targetValue is []string{value[:startPosiiton],value[startPosiiton:],""}
* otherwise the targetValue is []string{value[:startPosiiton],value[startPosiiton:endPosition],value[endPosition:]}

Reply:
Addressed.